File: backend/src/main/java/com/dheena/tasktracker/service/TaskService.java
package com.dheena.tasktracker.service;

import org.springframework.data.domain.Page;

import com.dheena.tasktracker.dto.CreateTaskRequest;
import com.dheena.tasktracker.dto.TaskResponse;
import com.dheena.tasktracker.dto.UpdateTaskRequest;
import com.dheena.tasktracker.entity.enums.TaskPriority;
import com.dheena.tasktracker.entity.enums.TaskStatus;

public interface TaskService {

    TaskResponse createTask(CreateTaskRequest request);

    TaskResponse updateTask(Long id, UpdateTaskRequest request);

    void deleteTask(Long id);

    TaskResponse getTaskById(Long id);

    Page<TaskResponse> getAllTasks(
            TaskStatus status,
            TaskPriority priority,
            int page,
            int size,
            String sortBy,
            String direction
    );
}

File: frontend/src/services/taskService.js
import api from "../api/axios";

export const getTasks = (
    status = "",
    priority = "",
    page = 0,
    size = 10,
    sortBy = "dueDate",
    direction = "asc"
) => {

    return api.get("/tasks", {
        params: {
            status,
            priority,
            page,
            size,
            sortBy,
            direction
        }
    });
};

export const getTask = (id) => api.get(`/tasks/${id}`);

export const createTask = (task) => api.post("/tasks", task);

export const updateTask = (id, task) =>
    api.put(`/tasks/${id}`, task);


export const deleteTask = (id) => api.delete(`/tasks/${id}`);